File: fixtures/adapter-java/basic/src/main/java/com/example/UserService.java
package com.example;

// Everything here is resolved: the analyzer follows imports, binds each
// receiver to its declared type, and walks supertypes to find an inherited
// method. What it cannot resolve — String, the standard library — records no
// edge rather than a guessed one.
public interface Auditable {
    String describe();
}

class AuditLog {
    void record(String message) {
    }
}

public class UserService implements Auditable {
    private AuditLog log;

    public String describe() {
        return "user service";
    }

    public void handle() {
        // A call to a method defined in this file.
        describe();
        // A call on a field, resolved through the field's declared type.
        log.record("handled");
    }
}

File: fixtures/adapter-csharp/basic/UserService.cs
// Everything here is resolved: the analyzer follows usings, binds each
// receiver to its declared type, and merges the halves of a partial class.
namespace Example
{
    public interface IAuditable
    {
        string Describe();
    }

    class AuditLog
    {
        public void Record(string message) { }
    }

    public class UserService : IAuditable
    {
        private AuditLog log;

        public string Describe()
        {
            return "user service";
        }

        public void Handle()
        {
            // A call to a method defined in this file.
            Describe();
            // A call on a field, resolved through the field's declared type.
            log.Record("handled");
        }
    }
}
